metricbeat/docs/autodiscover-kubernetes-config.asciidoc

```diff
@@ -67,7 +67,22 @@ where `REDIS_PASSWORD` is a key stored in local keystore of Metricbeat.
 [float]
 ===== Kubernetes Secrets
 Metricbeat autodiscover supports leveraging https://kubernetes.io/docs/concepts/configuration/secret/[Kubernetes secrets]
-in order to retrieve sensitive data like passwords.
+in order to retrieve sensitive data like passwords. In order to enable this future add the following section
+in Metricbeat's `ClusterRole` rules:
+
+["source","yaml",subs="attributes"]
+-------------------------------------------------------------------------------------
+- apiGroups: [""]
+  resources:
+    - secrets
+  verbs: ["get"]
+-------------------------------------------------------------------------------------
+
+CAUTION: The above rule will give permission to Metricbeat Pod to access Kubernetes Secrets API.
+This means that anyone who have access to Metricbeat Pod (`kubectl exec` for example) will be able to access
+Kubernetes Secrets API and get a specific secret no matter which namespace it belongs to. In this,
+this option should be carefully considered, specially when used with hints.
```

Comment:
Perhaps we can give a couple of examples about how to give permissions in specific namespaces (with role binding instead of cluster role binding), or to specific secrets (with `resourceNames` in the rule).

Or we could link to the kubernetes documentation: https://kubernetes.io/docs/reference/access-authn-authz/rbac/, so anyone can chose the approach that better fits with their needings.